strawberryfields/gbs/sample.py

```diff
@@ -332,37 +370,71 @@ def gaussian(
     if not 0 <= loss <= 1:
         raise ValueError("Loss parameter must take a value between zero and one")
 
-    n_modes = len(alpha)
+    n_modes = len(t)
+
+    # initialize engine
 
     eng = sf.LocalEngine(backend="gaussian")
-    gbs = sf.Program(n_modes)
 
+    # initialize program objects
+
+    if np.all(t != 0):
+        gbs = sf.Program(n_modes * 2)
+    else:
+        gbs = sf.Program(n_modes)
+
+    # start circuit
     # pylint: disable=expression-not-assigned,pointless-statement
     with gbs.context as q:
 
-        # this interferometer has no action, but is kept to follow the decomposition given in the
-        # docstring
-        sf.ops.Interferometer(U1) | q
+        # two-mode squeezing gates
+
+        if np.all(t != 0):
+            for i in range(n_modes):
+                sf.ops.S2gate(t[i]) | (q[i], q[i + n_modes])
+
+        # first interferometer
+
+        sf.ops.Interferometer(U1) | q[:n_modes]
+
+        # squeezing gates
 
         for i in range(n_modes):
             sf.ops.Sgate(r[i]) | q[i]
 
-        sf.ops.Interferometer(U2) | q
+        # second interferometer
+
+        sf.ops.Interferometer(U2) | q[:n_modes]
+
+        # displacement gates
 
         for i in range(n_modes):
             sf.ops.Dgate(alpha[i]) | q[i]
 
+        # apply loss
+
         if loss:
             for _q in q:
                 sf.ops.LossChannel(1 - loss) | _q
 
+        # measurement
+
         sf.ops.MeasureFock() | q
 
     with warnings.catch_warnings():
         warnings.filterwarnings("ignore", category=UserWarning, message="Cannot simulate non-")
+
+        # run the engine and generate samples
+
         s = eng.run(gbs, run_options={"shots": n_samples}).samples
 
+    # end circuit
+
     if n_samples == 1:
```

Comment:
If I understand correctly, the difference between a single sample or multiple samples is that a single sample is usually a list, whereas with many samples we have a list of lists. Would it be enough to just go
`if n_samples==1:`
    `s = [s]` 
and then use
 `if np.all(t == 0):`
        `s = np.pad(s, ((0, 0), (0, n_modes)))`?

Reply:
the pad syntax for a single sample and multiple samples is different. And the function should be able to pad both single and multiple samples with zeros.